File: packages/frontend/src/services/fornecedores.js
import axios from "axios";

const API_URL = "http://localhost:8000/fornecedores";

export async function listarFornecedores(limit, offset, busca) {
  const response = await axios.get(API_URL, {
    params: { limit, offset, busca }
  });
  return response.data;
}

export async function criarFornecedor(data) {
  const response = await axios.post(API_URL, data);
  return response.data;
}

export async function atualizarFornecedor(id, data) {
  const response = await axios.put(`${API_URL}/${id}`, data);
  return response.data;
}

export async function deletarFornecedor(id) {
  const response = await axios.delete(`${API_URL}/${id}`);
  return response.data;
}
export async function obterFornecedor(id) {
  const res = await axios.get(`${API_URL}/${id}`);
  return res.data;
}

File: packages/backend/app/api/fornecedores.py
from fastapi import APIRouter, HTTPException, Query
from app.services import fornecedor as fornecedor_service
from app.schemas.fornecedor import FornecedorCreate, FornecedorUpdate, FornecedorOut
from typing import List


router = APIRouter(prefix="/fornecedores", tags=["Fornecedores"])

@router.get("/", response_model=dict)
def listar_fornecedores(limit: int = 20, offset: int = 0, busca: str = ""):
    # Query no banco, aplicar busca/filtro e paginação
    results, total = fornecedor_service.buscar_fornecedores(limit=limit, offset=offset, busca=busca)
    return {"data": results, "total": total}

@router.get("/{fornecedor_id}", response_model=FornecedorOut)
def obter(fornecedor_id: int):
    result = fornecedor_service.get_by_id(fornecedor_id)
    if not result:
        raise HTTPException(status_code=404, detail="Fornecedor não encontrado")
    return result

@router.post("/", response_model=FornecedorOut)
def criar(fornecedor: FornecedorCreate):
    return fornecedor_service.create(fornecedor)

@router.put("/{fornecedor_id}", response_model=FornecedorOut)
def atualizar(fornecedor_id: int, fornecedor: FornecedorUpdate):
    result = fornecedor_service.update(fornecedor_id, fornecedor)
    if not result:
        raise HTTPException(status_code=404, detail="Fornecedor não encontrado")
    return result

@router.delete("/{fornecedor_id}")
def deletar(fornecedor_id: int):
    success = fornecedor_service.delete(fornecedor_id)
    if not success:
        raise HTTPException(status_code=404, detail="Fornecedor não encontrado")
    return {"ok": True}
